src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Win32.cs

```diff
@@ -736,6 +738,110 @@ private static bool TryCompareTimeZoneInformationToRegistry(in TIME_ZONE_INFORMA
             }
         }
 
+        /// <summary>
+        /// Helper function for getting the MUI path for a given resource and culture, using a cache to prevent duplicating work.
+        /// Searches the installed OS languages for either an exact matching culture, or one that has the same parent.
+        /// If not found, uses the preferred default OS UI language, to align with prior behavior.
+        /// </summary>
+        private static string TryGetCachedFileMuiPath(string filePath, CultureInfo cultureInfo)
+        {
+            string? result;
+            int hash = HashCode.Combine(filePath, cultureInfo);
```

Comment:
>hash [](http://example.com/codeflow?start=16&length=4)

is it possible we can run into collisions?

Reply:
I don't believe so.  Isn't that the point of `HashCode.Combine`?

Though, I just saw that `CultureInfo.GetHashCode` is implemented as `return Name.GetHashCode() + CompareInfo.GetHashCode();`.  Is that safe?

Ultimately, all I'm looking for here is a lookup for filePath+cultureInfo => fileMuiPath.  I could just concatenate them and use a `Dictionary<string,string>`, if you think that would be better.

Reply:
> I don't believe so.  Isn't that the point of HashCode.Combine?

I am not sure HashCode.Combine ever promised not generating colliding code. I know it is very unlikely but it is not impoossible. 

> Though, I just saw that CultureInfo.GetHashCode is implemented as return Name.GetHashCode() + CompareInfo.GetHashCode();.  Is that safe

safe against what? all hashing can possible generate colliding code. the difference is how likely this can happen. anyone using the hashing results should consider the collosion as possible to happen. 

> Ultimately, all I'm looking for here is a lookup for filePath+cultureInfo => fileMuiPath.  I could just concatenate them and use a Dictionary<string,string>, if you think that would be better.

I believe this will be better to leave the hashing to the dictionary to handle.

Reply:
Ok.  Switched to string cache key.